### PR TITLE
Port to Typescript, Prettier for linting

### PR DESCRIPTION
**Pull request by broofa**
'Been meaning to switch to TS for a while.  This provides some much-needed guard rails around the data structures being passed around.   E.g. the package / module structures we get from the npmjs repository.   While this isn't exactly rigorous (I only did a quick pass at defining the type structures there) it's a start, and provides something that can be iterated on.

In the midst of this work, I found myself struggling to get VSCode + typescript + parcel to all play nice with code linting and errors.   Using Prettier and the Prettier plugin in VSCode for formatting go that under control.  I know it makes for a lot of code churn, but the end result should be easier to maintain.

There's no functional changes here (other than 1-2 small existing bugs that Typescript identified)

### CODE REVIEW

File: .stylelintrc.json [hunk 1]
```diff
@@ -1,3 +0,0 @@
-{
-  "extends": "stylelint-config-standard"
-}
```
Comment (fregante) [inline .stylelintrc.json:3]: No more stylelint?

File: package.json [hunk 2]
```diff
@@ -42,26 +42,24 @@
...
     "standard-version": "9.3.0",
-    "stylelint": "13.13.1",
-    "stylelint-config-standard": "22.0.0"
+    "typescript": "4.4.4"
   },
   "scripts": {
     "build": "parcel build index.html",
     "release": "standard-version",
     "start": "parcel serve index.html",
-    "test": "eslint . && stylelint css",
```
Comment (fregante) [inline package.json:64]: Eslint is not the same as prettier though. Prettier is just formatting

**Review by fregante — APPROVED** [review]

File: .stylelintrc.json (hunk 1 above)
Comment (broofa) [reply .stylelintrc.json:3]: `prettier` [styles all the thingz](https://prettier.io/docs/en/index.html)

File: package.json (hunk 2 above)
Comment (broofa) [reply package.json:64]: Dooh!  Right you are.  Will fix.

File: .stylelintrc.json (hunk 1 above)
Comment (fregante) [reply .stylelintrc.json:3]: Style lint is *also* not just formatting 😃 

It’s ok though

File: package.json (hunk 2 above)
Comment (broofa) [reply package.json:64]: Holy fuck, trying to get vscode, eslint, prettier, typescript, react & JSX to all play nice ...
image

Comment (broofa) [reply package.json:64]: Well, that wasn't any trouble at all 🙄

File: .stylelintrc.json (hunk 1 above)
Comment (broofa) [reply .stylelintrc.json:3]: If I have to deal with another permutation of tooling at this point I'm gonna start pulling the wings off of butterflies.  🤯

Comment (broofa) [reply .stylelintrc.json:3]: Made a brief attempt at adding styleint back in (using `stylelint-prettier` and `stylelint-config-prettier`), but gave up when it started bitching about "When linting something other than CSS, you should install an appropriate syntax, e.g. "postcss-scss", and use the "customSyntax" option" after I'd told it to not lint the `.tsx` files.

I'm going to call this good for now.

**Pull request merged**